spec.md

```diff
@@ -1422,8 +1437,22 @@ message Error {
       // terminated or deleted before retrying.
       NODE_DOES_NOT_EXIST = 6;
 
+      // Indicates that the specified `NodeID` is not allowed or
+      // understood by the Plugin. More human-readable information MAY
+      // be provided in the `error_description` field.
+      //
+      // Recovery behavior: Caller MUST fix the `NodeID` before
+      // retrying.
       INVALID_NODE_ID = 7;
+
       VOLUME_NOT_ATTACHED_TO_SPECIFIED_NODE = 8;
+
+      // Indicates that the `NodeID` is not set to match that returned
```

Comment:
* If the `NodeID` is set, but it doesn't match `GetNodeID` then `INVALID_NODE_ID` should be returned.
* If `NodeID` is NOT set but this plugin RPC requires it, then `NODE_ID_REQUIRED` should be returned. This is meaningful because an empty `NodeID` in this case carries with it specific intentions.

.. so the first sentence of this description should be simply: "The plugin does not support the operation without a `NodeID`".